### ISSUE TITLE
Self Registration System

### ISSUE THREAD

**Issue by Sukhman-Baath** [pahulpreet-dev/Dooramo#6]
Adding new feature of automated self registration system for residents and service provider so that they can register can account themselves.

**Comment by haocencol** [comment]
Write this as a user story and put a "user story" label.

**Comment by haocencol** [comment]
A "New" label is redundant because Github time stamp all issues.

**Comment by Sukhman-Baath** [comment]
I have created two new Activities with name "ResidentSignUp" and "ResidentSignUp2". I have already done the design part for both. First activity will ask for details from resident and the second will ask to choose a username and password for login. I am confused about flow of logic for this module. We do not want our App to be used by some Non-Resident. Only residents under particular management should be able to use it. But, if we allow open sign up (for everyone), then anyone would be able to raise a request and then problem of redundant and fake requests would rise.

**Comment by pahulpreet-dev** [comment]
You are absolutely right @Sukhman-Baath about the redundant data. We do not want this. The working of this module will be as follow:
1. A person use the sign up form to provide details and select username and password for himself
2. As the person completes sign up, it will be updated in the database. We will have a new child node in the database for flag which will say initially say the person in not allowed to login.
3. The management console will have a new activity where the management team will receive requests for new sign ups.
4. The team will then decide whether to allow the person the login or not. If they allow, we will update the database flag which allows the person to login. If they deny login, we will remove the details from our database.

We need to update the Login module as well. In the login module, we will first look for the username, then we will check the newly created flag node which tells us if the person is allowed to login or not.

Also, the Service Provider Sign Up module will have same working.

If someone has better suggestion for the working, please share. @Sukhman-Baath @haocencol @AdavjeetBhangu @ramandhaliwal1605

**Comment by pahulpreet-dev** [comment]
done

**Comment by pahulpreet-dev** [comment]
The team did a great job re-using all the resources we already had. The only new class we created was the Java beans class for helper purposes. We used the same Create user and create service provider activities to save resources. To distinguish between who opened the activity, we used flags. Similarly, to store the data we checked the same flags. The data stored in the newly created child node will be "pending" (verification) if the self registration feature is used and "done" (verification) if management is creating the new users. Further, management can see all the verification pending requests in a new activity and the decide whether to accept the registration request or drop it.
